File: RecentMistakesFirstSorter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecentMistakesFirstSorter = void 0;
class RecentMistakesFirstSorter {
    reorganize(cards) {
        if (!cards || cards.length <= 1) {
            return cards;
        }
        const incorrect = [];
        const others = [];
        for (const card of cards) {
            if (card.wasIncorrectInLastRound()) {
                incorrect.push(card);
            }
            else {
                others.push(card);
            }
        }
        return [...incorrect, ...others];
    }
}
exports.RecentMistakesFirstSorter = RecentMistakesFirstSorter;

File: RecentMistakesFirstSorter.java
import java.util.ArrayList;
import java.util.List;

public class RecentMistakesFirstSorter implements CardOrganizer {

    @Override
    public List<Card> reorganize(List<Card> cards) {
        if (cards == null || cards.size() <= 1) {
            return cards;
        }

        List<Card> incorrect = new ArrayList<>();
        List<Card> others = new ArrayList<>();

        for (Card card : cards) {
            if (card.wasIncorrectInLastRound()) {
                incorrect.add(card);
            } else {
                others.add(card);
            }
        }

        List<Card> result = new ArrayList<>(cards.size());
        result.addAll(incorrect);
        result.addAll(others);

        return result;
    }
}